[2024-06-26T19:38:54.650+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-06-26T19:38:54.692+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: Stage_1.run_bronze_job manual__2024-06-26T19:38:47.976496+00:00 [queued]>
[2024-06-26T19:38:54.704+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: Stage_1.run_bronze_job manual__2024-06-26T19:38:47.976496+00:00 [queued]>
[2024-06-26T19:38:54.704+0000] {taskinstance.py:2306} INFO - Starting attempt 1 of 1
[2024-06-26T19:38:54.719+0000] {taskinstance.py:2330} INFO - Executing <Task(DockerOperator): run_bronze_job> on 2024-06-26 19:38:47.976496+00:00
[2024-06-26T19:38:54.728+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=523) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-06-26T19:38:54.730+0000] {standard_task_runner.py:63} INFO - Started process 524 to run task
[2024-06-26T19:38:54.730+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'Stage_1', 'run_bronze_job', 'manual__2024-06-26T19:38:47.976496+00:00', '--job-id', '80', '--raw', '--subdir', 'DAGS_FOLDER/kafka_dag.py', '--cfg-path', '/tmp/tmppclkea21']
[2024-06-26T19:38:54.732+0000] {standard_task_runner.py:91} INFO - Job 80: Subtask run_bronze_job
[2024-06-26T19:38:54.748+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/settings.py:195 DeprecationWarning: The sql_alchemy_conn option in [core] has been moved to the sql_alchemy_conn option in [database] - the old setting has been used, but please update your config.
[2024-06-26T19:38:54.787+0000] {task_command.py:426} INFO - Running <TaskInstance: Stage_1.run_bronze_job manual__2024-06-26T19:38:47.976496+00:00 [running]> on host ba36c0353089
[2024-06-26T19:38:54.892+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='Stage_1' AIRFLOW_CTX_TASK_ID='run_bronze_job' AIRFLOW_CTX_EXECUTION_DATE='2024-06-26T19:38:47.976496+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-06-26T19:38:47.976496+00:00'
[2024-06-26T19:38:54.892+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-06-26T19:38:54.928+0000] {docker.py:366} INFO - Starting docker container from image bitnami/spark:latest
[2024-06-26T19:38:54.931+0000] {docker.py:374} WARNING - Using remote engine or docker-in-docker and mounting temporary volume from host is not supported. Falling back to `mount_tmp_dir=False` mode. You can set `mount_tmp_dir` parameter to False to disable mounting and remove the warning
[2024-06-26T19:38:55.538+0000] {docker.py:436} INFO - [38;5;6mspark [38;5;5m19:38:55.53 [0m[38;5;2mINFO [0m ==>
[2024-06-26T19:38:55.542+0000] {docker.py:436} INFO - [38;5;6mspark [38;5;5m19:38:55.53 [0m[38;5;2mINFO [0m ==> [1mWelcome to the Bitnami spark container[0m
[2024-06-26T19:38:55.542+0000] {docker.py:436} INFO - [38;5;6mspark [38;5;5m19:38:55.54 [0m[38;5;2mINFO [0m ==> Subscribe to project updates by watching [1mhttps://github.com/bitnami/containers[0m
[2024-06-26T19:38:55.546+0000] {docker.py:436} INFO - [38;5;6mspark [38;5;5m19:38:55.54 [0m[38;5;2mINFO [0m ==> Submit issues and feature requests at [1mhttps://github.com/bitnami/containers/issues[0m
[2024-06-26T19:38:55.549+0000] {docker.py:436} INFO - [38;5;6mspark [38;5;5m19:38:55.54 [0m[38;5;2mINFO [0m ==> Upgrade to Tanzu Application Catalog for production environments to access custom-configured and pre-packaged software components. Gain enhanced features, including Software Bill of Materials (SBOM), CVE scan result reports, and VEX documents. To learn more, visit [1mhttps://bitnami.com/enterprise[0m
[2024-06-26T19:38:55.552+0000] {docker.py:436} INFO - [38;5;6mspark [38;5;5m19:38:55.55 [0m[38;5;2mINFO [0m ==>
[2024-06-26T19:38:55.563+0000] {docker.py:436} INFO - 
[2024-06-26T19:38:57.905+0000] {docker.py:436} INFO - :: loading settings :: url = jar:file:/opt/bitnami/spark/jars/ivy-2.5.1.jar!/org/apache/ivy/core/settings/ivysettings.xml
[2024-06-26T19:38:57.995+0000] {docker.py:436} INFO - Ivy Default Cache set to: /opt/bitnami/spark/.ivy2/cache
[2024-06-26T19:38:57.996+0000] {docker.py:436} INFO - The jars for the packages stored in: /opt/bitnami/spark/.ivy2/jars
[2024-06-26T19:38:58.004+0000] {docker.py:436} INFO - org.apache.spark#spark-sql-kafka-0-10_2.12 added as a dependency
[2024-06-26T19:38:58.005+0000] {docker.py:436} INFO - org.postgresql#postgresql added as a dependency
[2024-06-26T19:38:58.008+0000] {docker.py:436} INFO - :: resolving dependencies :: org.apache.spark#spark-submit-parent-ec8eb0fe-c656-4d1a-8b81-5cf308994ab8;1.0
[2024-06-26T19:38:58.009+0000] {docker.py:436} INFO - confs: [default]
[2024-06-26T19:39:00.172+0000] {docker.py:436} INFO - found org.apache.spark#spark-sql-kafka-0-10_2.12;3.3.0 in central
[2024-06-26T19:39:00.660+0000] {docker.py:436} INFO - found org.apache.spark#spark-token-provider-kafka-0-10_2.12;3.3.0 in central
[2024-06-26T19:39:00.783+0000] {docker.py:436} INFO - found org.apache.kafka#kafka-clients;2.8.1 in central
[2024-06-26T19:39:00.966+0000] {docker.py:436} INFO - found org.lz4#lz4-java;1.8.0 in central
[2024-06-26T19:39:01.078+0000] {docker.py:436} INFO - found org.xerial.snappy#snappy-java;1.1.8.4 in central
[2024-06-26T19:39:02.167+0000] {docker.py:436} INFO - found org.slf4j#slf4j-api;1.7.32 in central
[2024-06-26T19:39:02.838+0000] {docker.py:436} INFO - found org.apache.hadoop#hadoop-client-runtime;3.3.2 in central
[2024-06-26T19:39:03.065+0000] {docker.py:436} INFO - found org.spark-project.spark#unused;1.0.0 in central
[2024-06-26T19:39:03.279+0000] {docker.py:436} INFO - found org.apache.hadoop#hadoop-client-api;3.3.2 in central
[2024-06-26T19:39:03.606+0000] {docker.py:436} INFO - found commons-logging#commons-logging;1.1.3 in central
[2024-06-26T19:39:03.694+0000] {docker.py:436} INFO - found com.google.code.findbugs#jsr305;3.0.0 in central
[2024-06-26T19:39:04.740+0000] {docker.py:436} INFO - found org.apache.commons#commons-pool2;2.11.1 in central
[2024-06-26T19:39:06.696+0000] {docker.py:436} INFO - found org.postgresql#postgresql;42.2.2 in central
[2024-06-26T19:39:06.739+0000] {docker.py:436} INFO - downloading https://repo1.maven.org/maven2/org/apache/spark/spark-sql-kafka-0-10_2.12/3.3.0/spark-sql-kafka-0-10_2.12-3.3.0.jar ...
[2024-06-26T19:39:06.843+0000] {docker.py:436} INFO - [SUCCESSFUL ] org.apache.spark#spark-sql-kafka-0-10_2.12;3.3.0!spark-sql-kafka-0-10_2.12.jar (122ms)
[2024-06-26T19:39:06.872+0000] {docker.py:436} INFO - downloading https://repo1.maven.org/maven2/org/postgresql/postgresql/42.2.2/postgresql-42.2.2.jar ...
[2024-06-26T19:39:07.045+0000] {docker.py:436} INFO - [SUCCESSFUL ] org.postgresql#postgresql;42.2.2!postgresql.jar(bundle) (201ms)
[2024-06-26T19:39:07.066+0000] {docker.py:436} INFO - downloading https://repo1.maven.org/maven2/org/apache/spark/spark-token-provider-kafka-0-10_2.12/3.3.0/spark-token-provider-kafka-0-10_2.12-3.3.0.jar ...
[2024-06-26T19:39:07.114+0000] {docker.py:436} INFO - [SUCCESSFUL ] org.apache.spark#spark-token-provider-kafka-0-10_2.12;3.3.0!spark-token-provider-kafka-0-10_2.12.jar (67ms)
[2024-06-26T19:39:07.136+0000] {docker.py:436} INFO - downloading https://repo1.maven.org/maven2/org/apache/kafka/kafka-clients/2.8.1/kafka-clients-2.8.1.jar ...
[2024-06-26T19:39:07.931+0000] {docker.py:436} INFO - [SUCCESSFUL ] org.apache.kafka#kafka-clients;2.8.1!kafka-clients.jar (814ms)
[2024-06-26T19:39:07.957+0000] {docker.py:436} INFO - downloading https://repo1.maven.org/maven2/com/google/code/findbugs/jsr305/3.0.0/jsr305-3.0.0.jar ...
[2024-06-26T19:39:07.981+0000] {docker.py:436} INFO - [SUCCESSFUL ] com.google.code.findbugs#jsr305;3.0.0!jsr305.jar (48ms)
[2024-06-26T19:39:08.013+0000] {docker.py:436} INFO - downloading https://repo1.maven.org/maven2/org/apache/commons/commons-pool2/2.11.1/commons-pool2-2.11.1.jar ...
[2024-06-26T19:39:08.063+0000] {docker.py:436} INFO - [SUCCESSFUL ] org.apache.commons#commons-pool2;2.11.1!commons-pool2.jar (80ms)
[2024-06-26T19:39:08.094+0000] {docker.py:436} INFO - downloading https://repo1.maven.org/maven2/org/spark-project/spark/unused/1.0.0/unused-1.0.0.jar ...
[2024-06-26T19:39:08.432+0000] {docker.py:436} INFO - [SUCCESSFUL ] org.spark-project.spark#unused;1.0.0!unused.jar (368ms)
[2024-06-26T19:39:08.465+0000] {docker.py:436} INFO - downloading https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-client-runtime/3.3.2/hadoop-client-runtime-3.3.2.jar ...
[2024-06-26T19:39:11.384+0000] {docker.py:436} INFO - [SUCCESSFUL ] org.apache.hadoop#hadoop-client-runtime;3.3.2!hadoop-client-runtime.jar (2951ms)
[2024-06-26T19:39:11.424+0000] {docker.py:436} INFO - downloading https://repo1.maven.org/maven2/org/lz4/lz4-java/1.8.0/lz4-java-1.8.0.jar ...
[2024-06-26T19:39:11.504+0000] {docker.py:436} INFO - [SUCCESSFUL ] org.lz4#lz4-java;1.8.0!lz4-java.jar (117ms)
[2024-06-26T19:39:11.530+0000] {docker.py:436} INFO - downloading https://repo1.maven.org/maven2/org/xerial/snappy/snappy-java/1.1.8.4/snappy-java-1.1.8.4.jar ...
[2024-06-26T19:39:11.816+0000] {docker.py:436} INFO - [SUCCESSFUL ] org.xerial.snappy#snappy-java;1.1.8.4!snappy-java.jar(bundle) (309ms)
[2024-06-26T19:39:11.836+0000] {docker.py:436} INFO - downloading https://repo1.maven.org/maven2/org/slf4j/slf4j-api/1.7.32/slf4j-api-1.7.32.jar ...
[2024-06-26T19:39:11.881+0000] {docker.py:436} INFO - [SUCCESSFUL ] org.slf4j#slf4j-api;1.7.32!slf4j-api.jar (64ms)
[2024-06-26T19:39:11.909+0000] {docker.py:436} INFO - downloading https://repo1.maven.org/maven2/org/apache/hadoop/hadoop-client-api/3.3.2/hadoop-client-api-3.3.2.jar ...
[2024-06-26T19:39:13.715+0000] {docker.py:436} INFO - [SUCCESSFUL ] org.apache.hadoop#hadoop-client-api;3.3.2!hadoop-client-api.jar (1833ms)
[2024-06-26T19:39:13.736+0000] {docker.py:436} INFO - downloading https://repo1.maven.org/maven2/commons-logging/commons-logging/1.1.3/commons-logging-1.1.3.jar ...
[2024-06-26T19:39:13.763+0000] {docker.py:436} INFO - [SUCCESSFUL ] commons-logging#commons-logging;1.1.3!commons-logging.jar (46ms)
[2024-06-26T19:39:13.763+0000] {docker.py:436} INFO - :: resolution report :: resolve 8712ms :: artifacts dl 7045ms
[2024-06-26T19:39:13.764+0000] {docker.py:436} INFO - :: modules in use:
[2024-06-26T19:39:13.764+0000] {docker.py:436} INFO - com.google.code.findbugs#jsr305;3.0.0 from central in [default]
	commons-logging#commons-logging;1.1.3 from central in [default]
	org.apache.commons#commons-pool2;2.11.1 from central in [default]
[2024-06-26T19:39:13.765+0000] {docker.py:436} INFO - org.apache.hadoop#hadoop-client-api;3.3.2 from central in [default]
[2024-06-26T19:39:13.765+0000] {docker.py:436} INFO - org.apache.hadoop#hadoop-client-runtime;3.3.2 from central in [default]
[2024-06-26T19:39:13.766+0000] {docker.py:436} INFO - org.apache.kafka#kafka-clients;2.8.1 from central in [default]
[2024-06-26T19:39:13.766+0000] {docker.py:436} INFO - org.apache.spark#spark-sql-kafka-0-10_2.12;3.3.0 from central in [default]
[2024-06-26T19:39:13.767+0000] {docker.py:436} INFO - org.apache.spark#spark-token-provider-kafka-0-10_2.12;3.3.0 from central in [default]
[2024-06-26T19:39:13.767+0000] {docker.py:436} INFO - org.lz4#lz4-java;1.8.0 from central in [default]
[2024-06-26T19:39:13.768+0000] {docker.py:436} INFO - org.postgresql#postgresql;42.2.2 from central in [default]
[2024-06-26T19:39:13.768+0000] {docker.py:436} INFO - org.slf4j#slf4j-api;1.7.32 from central in [default]
[2024-06-26T19:39:13.769+0000] {docker.py:436} INFO - org.spark-project.spark#unused;1.0.0 from central in [default]
[2024-06-26T19:39:13.769+0000] {docker.py:436} INFO - org.xerial.snappy#snappy-java;1.1.8.4 from central in [default]
[2024-06-26T19:39:13.770+0000] {docker.py:436} INFO - ---------------------------------------------------------------------
[2024-06-26T19:39:13.771+0000] {docker.py:436} INFO - |                  |            modules            ||   artifacts   |
[2024-06-26T19:39:13.771+0000] {docker.py:436} INFO - |       conf       | number| search|dwnlded|evicted|| number|dwnlded|
	---------------------------------------------------------------------
[2024-06-26T19:39:13.772+0000] {docker.py:436} INFO - |      default     |   13  |   13  |   13  |   0   ||   13  |   13  |
	---------------------------------------------------------------------
[2024-06-26T19:39:13.776+0000] {docker.py:436} INFO - :: retrieving :: org.apache.spark#spark-submit-parent-ec8eb0fe-c656-4d1a-8b81-5cf308994ab8
[2024-06-26T19:39:13.777+0000] {docker.py:436} INFO - confs: [default]
[2024-06-26T19:39:13.940+0000] {docker.py:436} INFO - 13 artifacts copied, 0 already retrieved (57403kB/164ms)
[2024-06-26T19:39:14.257+0000] {docker.py:436} INFO - 24/06/26 19:39:14 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-06-26T19:39:15.746+0000] {docker.py:436} INFO - Traceback (most recent call last):
  File "/opt/bitnami/spark/scripts/spark_bronze_layer.py", line 10, in <module>
    from pyspark.sql.functions import from_json, col,F
ImportError: cannot import name 'F' from 'pyspark.sql.functions' (/opt/bitnami/spark/python/lib/pyspark.zip/pyspark/sql/functions.py)
[2024-06-26T19:39:15.814+0000] {docker.py:436} INFO - 24/06/26 19:39:15 INFO ShutdownHookManager: Shutdown hook called
[2024-06-26T19:39:15.815+0000] {docker.py:436} INFO - 24/06/26 19:39:15 INFO ShutdownHookManager: Deleting directory /tmp/spark-15b6219d-d712-4949-bbdc-ee35fe4b0772
[2024-06-26T19:39:18.270+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-06-26T19:39:18.284+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/docker/api/client.py", line 265, in _raise_for_status
    response.raise_for_status()
  File "/home/airflow/.local/lib/python3.12/site-packages/requests/models.py", line 1021, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 400 Client Error: Bad Request for url: http://docker-proxy:2375/v1.45/containers/create

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 371, in _run_image
    return self._run_image_with_mounts([*self.mounts, tmp_mount], add_tmp_variable=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 398, in _run_image_with_mounts
    self.container = self.cli.create_container(
                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/docker/api/container.py", line 439, in create_container
    return self.create_container_from_config(config, name, platform)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/docker/api/container.py", line 456, in create_container_from_config
    return self._result(res, True)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/docker/api/client.py", line 271, in _result
    self._raise_for_status(response)
  File "/home/airflow/.local/lib/python3.12/site-packages/docker/api/client.py", line 267, in _raise_for_status
    raise create_api_error_from_http_exception(e) from e
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/docker/errors.py", line 39, in create_api_error_from_http_exception
    raise cls(e, response=response, explanation=explanation) from e
docker.errors.APIError: 400 Client Error for http://docker-proxy:2375/v1.45/containers/create: Bad Request ("invalid mount config for type "bind": bind source path does not exist: /tmp/airflowtmphqvqge2l")

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 509, in execute
    return self._run_image()
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 380, in _run_image
    return self._run_image_with_mounts(self.mounts, add_tmp_variable=False)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/providers/docker/operators/docker.py", line 444, in _run_image_with_mounts
    raise DockerContainerFailedException(f"Docker container failed: {result!r}", logs=log_lines)
airflow.providers.docker.exceptions.DockerContainerFailedException: Docker container failed: {'StatusCode': 1}
[2024-06-26T19:39:18.296+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=Stage_1, task_id=run_bronze_job, run_id=manual__2024-06-26T19:38:47.976496+00:00, execution_date=20240626T193847, start_date=20240626T193854, end_date=20240626T193918
[2024-06-26T19:39:18.315+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 80 for task run_bronze_job (Docker container failed: {'StatusCode': 1}; 524)
[2024-06-26T19:39:18.348+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-06-26T19:39:18.378+0000] {taskinstance.py:3498} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-06-26T19:39:18.381+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
